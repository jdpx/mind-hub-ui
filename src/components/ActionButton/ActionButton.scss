@import "../../styles/colours.scss";

.btn-wrapper {
    display: flex;
    align-items: center;
    margin-left: 15px;
    margin-right: 15px;

    &.center {
        justify-content: center;
    }

    &.left {
        justify-content: flex-start;
    }

    &.right {
        justify-content: flex-end;
    }
}

.btn {
    font-weight: bold;
    font-size: 20px;
    line-height: 22px;
    text-transform: capitalize;
    padding: 10px 40px;

    &.primary {
        background-color: $orange;
        border-radius: 20px;

        &:hover {
            color: $white;
            background-color: $mild-orange;
        }

        &:active {
            background-color: $dark-orange;
        }
    }
    &.secondary {
        color: $white;
        background-color: $light-grey;
        border-radius: 20px;
    }
}
