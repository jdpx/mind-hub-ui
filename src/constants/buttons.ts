export const Primary = 'primary'
export const Secondary = 'secondary'
