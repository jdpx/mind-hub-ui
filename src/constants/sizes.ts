export const Small = 'sm'
export const Medium = 'md'
export const Large = 'lg'
export const ExtraLarge = 'xl'
