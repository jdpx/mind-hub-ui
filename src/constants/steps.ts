export const Video = 'Video'
export const Audio = 'Audio'
export const Question = 'Question'
