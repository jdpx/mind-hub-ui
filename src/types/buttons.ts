import { Center, Left, Primary, Right, Secondary } from '../constants/buttons'

export type ButtonTypes = typeof Primary | typeof Secondary

export type Positions = typeof Right | typeof Left | typeof Center
