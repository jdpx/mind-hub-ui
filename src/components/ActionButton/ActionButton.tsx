import React from 'react'
import classNames from 'classnames'

import { Center, Primary } from '../../constants/buttons'
import { ButtonTypes, Positions } from '../../types/buttons'

import './ActionButton.scss'

interface Props {
    text: string
    position?: Positions
    type?: ButtonTypes
    disabled?: boolean
    onClick?: () => void
    testid?: string
}

export default function ActionButton({
    text,
    position = Center,
    type = Primary,
    disabled,
    onClick,
    testid,
}: Props) {
    const buttonClass = classNames('btn', type, {
        disabled: disabled,
    })

    const wrapperClass = classNames('btn-wrapper', position)

    return (
        <div className={wrapperClass} data-testid={testid}>
            <button
                className={buttonClass}
                onClick={onClick}
                disabled={disabled}
                data-testid={`${testid}-btn`}
            >
                {text}
            </button>
        </div>
    )
}
