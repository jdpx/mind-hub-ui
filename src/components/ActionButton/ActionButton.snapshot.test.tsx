import React from 'react'
import { render } from '@testing-library/react'
import ActionButton from './ActionButton'
import { Secondary } from '../../constants/buttons'

describe('ActionButton Snapshots', () => {
    const props = {
        text: 'foo',
    }

    it('renders action button', () => {
        const { asFragment } = render(<ActionButton {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders secondary type button', () => {
        const { asFragment } = render(<ActionButton type={Secondary} {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders a disabled button', () => {
        const { asFragment } = render(<ActionButton disabled {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders button with testid', () => {
        const { asFragment } = render(<ActionButton testid="action-button" {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })
})
