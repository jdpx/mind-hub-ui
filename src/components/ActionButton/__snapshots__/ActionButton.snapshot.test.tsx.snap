// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Snapshots renders a disabled button 1`] = `
<DocumentFragment>
  <div
    class="btn-wrapper center"
  >
    <button
      class="btn primary disabled"
      data-testid="undefined-btn"
      disabled=""
    >
      foo
    </button>
  </div>
</DocumentFragment>
`;

exports[`ActionButton Snapshots renders action button 1`] = `
<DocumentFragment>
  <div
    class="btn-wrapper center"
  >
    <button
      class="btn primary"
      data-testid="undefined-btn"
    >
      foo
    </button>
  </div>
</DocumentFragment>
`;

exports[`ActionButton Snapshots renders button with testid 1`] = `
<DocumentFragment>
  <div
    class="btn-wrapper center"
    data-testid="action-button"
  >
    <button
      class="btn primary"
      data-testid="action-button-btn"
    >
      foo
    </button>
  </div>
</DocumentFragment>
`;

exports[`ActionButton Snapshots renders secondary type button 1`] = `
<DocumentFragment>
  <div
    class="btn-wrapper center"
  >
    <button
      class="btn secondary"
      data-testid="undefined-btn"
    >
      foo
    </button>
  </div>
</DocumentFragment>
`;
